app/templates/ui/app/route/routes.js

```diff
@@ -86,6 +86,12 @@
             return MLRest.getDocument(uri, { format: 'json' }).then(function(response) {
               return response.data;
             });
+          },
+          type: function(MLRest, $stateParams) {
+            var uri = $stateParams.uri;
+            return MLRest.extension('exttype', { method: 'GET', params: { 'rs:uri': uri}}).then(function(response) {
```

Comment:
Also thinking, can't we somehow append the mime-type/content-type to the getDocument above? Perhaps add a Content-type response-header (if not present yet) using a transform?